File: 15-word-freq/word_freq.exs
# Elixir program to count the no of occurances of characters
# Written by M.V.Harish Kumar on 14-01-2024
s = IO.gets "Enter a string: "
IO.puts "The no of occurances are:"
s
|> String.trim
|> String.graphemes
|> Enum.frequencies
|> Enum.each(fn {k, v} -> IO.puts "\t'#{k}': #{v}" end)

File: 15-word-freq/word_freq.rb
# Ruby program to count letter frequency
# Written by M.V.Harish Kumar on 01-02-2025

print "Enter a string: "
str = gets.chomp
freq = {}
str.each_char do |ch|
  if freq.include? ch
    freq[ch] += 1
  else
    freq[ch] = 1
  end
end

puts "The no of occurances are:"
freq.each do |ch, cnt|
  puts "\t#{ch.inspect}: #{cnt}"
end
